techmatters/webchat
# Changes to End chat functionality when task is assigned stage

## Description
- This PR along with serverless [PR](https://github.com/techmatters/serverless/pull/341) has minor refactor to check task at assigned stage for triggering end chat more reliably
- Adds `isTaskStageAssigned` as response field for endChat and sends automated message only in case of task in assigned stage

### Checklist

### Related Issues
Fixes #[CHI-1402](https://bugs.benetech.org/browse/CHI-1402)

### Verification steps
- Ensure that this [PR](https://github.com/techmatters/serverless/pull/341) for serverless is deployed to Aselo production.
- Try 'End Chat' when the task is in the following states: pending, reserved, assigned. In the case of assigned, a message that user ended chat should be received in flex and webchat
